### PR TITLE
[release-4.5] Bug 1885936: Gather ServiceAccounts

### PR DESCRIPTION
**Pull request by rluders**
This PR backports IO enhancement to gather ServiceAccounts stats from cluster namespaces.

## Categories

- [ ] Bugfix
- [x] Enhancement
- [X] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample archive

- `docs/insights-archive-sample/config/serviceaccounts.json`

## Documentation

- `docs/gathered-data.md`

## Unit Tests

- `pkg/gather/clusterconfig/clusterconfig_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## References

https://issues.redhat.com/browse/CCXDEV-3250
https://bugzilla.redhat.com/show_bug.cgi?id=1885936
https://issues.redhat.com/browse/INSIGHTOCP-75

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,11 @@
+GO_PKG=github.com/openshift/insights-operator
+GO111MODULE?=on
+GOFLAGS=-mod=vendor
+export GO111MODULE
+export GOFLAGS
+
 build:
-	GO111MODULE=on go build -mod=vendor -ldflags "-X k8s.io/client-go/pkg/version.gitCommit=$$(git rev-parse HEAD) -X k8s.io/client-go/pkg/version.gitVersion=v1.0.0+$$(git rev-parse --short=7 HEAD)" -o bin/insights-operator ./cmd/insights-operator
+	GO111MODULE=on go build -ldflags "-X $(GO_PKG)/vendor/k8s.io/client-go/pkg/version.gitCommit=$$(git rev-parse HEAD) -X $(GO_PKG)/vendor/k8s.io/client-go/pkg/version.gitVersion=v1.0.0+$$(git rev-parse --short=7 HEAD)" -o bin/insights-operator ./cmd/insights-operator
```
Comment (0sewa0) [inline Makefile:8]: This is actually incorrect, had the same issue in my PR that I had to rebase (https://github.com/openshift/insights-operator/pull/244)

Comment (0sewa0) [reply Makefile:8]: And this was the actual merge conflict, introduced by https://github.com/openshift/insights-operator/pull/204

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,9 @@
+GO_PKG=github.com/openshift/insights-operator
+GO111MODULE?=on
+GOFLAGS=-mod=vendor
+export GO111MODULE
+export GOFLAGS
+
```
Comment (0sewa0) [inline Makefile:6]: I don't believe these are needed, I had the same on one of my backports

Comment (rluders) [reply Makefile:6]: Now that you pointed it all, I think that the only thing that makes sense to keep is the: `GO111MODULE?=on`

Comment (rluders) [reply Makefile:6]: Or maybe not even this one... o_O

**Comment by 0sewa0** [comment]
tested with cluster-bot, works as expected
/lgtm

**Review by natiiix — APPROVED** [review]
Just a rebase, nothing has changed since the last approval.
/lgtm

**Pull request merged**